### PR TITLE
Enable Skia's Vulkan backend on all non-mobile platforms, and test SwiftShader Vulkan on all GL unittests platforms

### PR DESCRIPTION
**Pull request by bdero**
- Enables Skia's Vulkan backend on all platforms (which will be usable through the embedder API).
- Gets SwiftShader Vulkan building anywhere the GL embedder unittests can run.
- Adds a simple test that initializes Vulkan state with SwiftShader (full test harness integration to come).

Patches that need to land before this is merged:
* Skia: https://skia-review.googlesource.com/c/skia/+/469207/
* buildroot: https://github.com/flutter/buildroot/pull/524

### CODE REVIEW

**Comment by bdero** [comment]
Unfortunately, Skia's Vulkan backend can't be built for iOS 9 (the version we target) right now because Skia uses VMA, and VMA uses C++17 standard library features not available prior to iOS 10.

File: testing/test_vulkan_context.h
Comment (chinmaygarde) [inline testing/test_vulkan_context.h:20]: Nit: `bool IsValid() const;`.

The case because its OOL and the `const` because of const-correctness.

File: testing/test_vulkan_context.h
Comment (chinmaygarde) [inline testing/test_vulkan_context.h:27]: `FML_DISALLOW_COPY_AND_ASSIGN` explicitly even though this can't really be copied because of the unique pointers.

File: testing/test_vulkan_context.h
Comment (chinmaygarde) [inline testing/test_vulkan_context.h:23]: Default to false here in case we add additional ctors and forget to initialize the scalar explicitly in one of them? Your call.

File: vulkan/vulkan_proc_table.cc
Comment (chinmaygarde) [inline vulkan/vulkan_proc_table.cc:160]: lol, I never thought of this to avoid an explicit bool cast without actually using an explicit cast. Love it.

File: tools/gn
Comment (chinmaygarde) [inline tools/gn:313]: I believe this will compile the vulkan shell component even on iOS and Android. The desire is to compile Vulkan on all host unit-test targets right?

**Review by chinmaygarde — COMMENTED** [review]
Mostly LGTM except I don't think setting shell_enable_vulkan to true unconditionally is what we want for iOS and Android.

File: tools/gn
Comment (bdero) [reply tools/gn:313]: My intention was to enable Vulkan support for all libflutter platform distributions including iOS, so that embedder writers have the option to target just Vulkan and use translation ICDs like MoltenVK. Partially because I want it, but also just because we can. :)

Comment (chinmaygarde) [reply tools/gn:313]: With that, you run into the issue of bumping up the binary size on those target. And we are pretty sensitive to that metric.

Comment (bdero) [reply tools/gn:313]: I'll disable this for Android/iOS for the time being and leave a note since it's easier to flip on than flip off later.

File: testing/test_vulkan_context.h
Comment (bdero) [reply testing/test_vulkan_context.h]: Done.

File: testing/test_vulkan_context.h
Comment (bdero) [reply testing/test_vulkan_context.h:29]: Done.

File: testing/test_vulkan_context.h
Comment (bdero) [reply testing/test_vulkan_context.h]: sgtm, done.

**Comment by bdero** [comment]
This is waiting on the next Skia roll to land at the moment.

**Comment by bdero** [comment]
@chinmaygarde Could you give this another look when you get a moment? I think this should be good to go.

**Review by chinmaygarde — APPROVED** [review]

**Comment by knopp** [comment]
How is this supposed to work on macos? I get build error when it tries to compile `//fluter/vulkan`, possibly as part of open gl tests.

**Comment by bdero** [comment]
@knopp The embedder unittests link Swiftshader, which is one of several Vulkan ICDs that support MacOS. Could you open an issue with the error you're getting/repro steps and tag me on it?

**Comment by knopp** [comment]
@bdero, nevermind, I think the problem was missing `skia_use_vulkan` in `args.gn`.

**Pull request merged**
